assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidgetV2/index.js

```diff
@@ -20,31 +20,101 @@
 import { CORE_SITE } from '../../../../../googlesitekit/datastore/site/constants';
+import { CORE_USER } from '../../../../../googlesitekit/datastore/user/constants';
+import { Grid, Row, Cell } from '../../../../../material-components/layout';
+import { getURLPath } from '../../../../../util/getURLPath';
 import whenActive from '../../../../../util/when-active';
+import SourceLink from '../../../../../components/SourceLink';
+import ReportError from '../../../../../components/ReportError';
+import ReportZero from '../../../../../components/ReportZero';
 import TotalUserCount from './TotalUserCount';
 import UserCountGraph from './UserCountGraph';
 import DimensionTabs from './DimensionTabs';
 import UserDimensionsPieChart from './UserDimensionsPieChart';
-import SourceLink from '../../../../../components/SourceLink';
-import { Grid, Row, Cell } from '../../../../../material-components/layout';
-import { getURLPath } from '../../../../../util/getURLPath';
 const { Widget } = Widgets.components;
 const { useSelect } = Data;
 
 function DashboardAllTrafficWidget() {
+	const [ firstLoad, setFirstLoad ] = useState( true );
+	const [ currentRange, setCurrentRange ] = useState( '' );
+
+	const dateRange = useSelect( ( select ) => select( CORE_USER ).getDateRange() );
 	const dimensionName = useSelect( ( select ) => select( CORE_FORMS ).getValue( FORM_ALL_TRAFFIC_WIDGET, 'dimensionName' ) || 'ga:channelGrouping' );
 	const dimensionValue = useSelect( ( select ) => select( CORE_FORMS ).getValue( FORM_ALL_TRAFFIC_WIDGET, 'dimensionValue' ) );
 	const entityURL = useSelect( ( select ) => select( CORE_SITE ).getCurrentEntityURL() );
 
+	const {
+		startDate,
+		endDate,
+		compareStartDate,
+		compareEndDate,
+	} = useSelect( ( select ) => select( CORE_USER ).getDateRangeDates( {
+		compare: true,
+		offsetDays: DATE_RANGE_OFFSET,
+	} ) );
+
+	const baseArgs = {
+		startDate,
+		endDate,
+		metrics: [ { expression: 'ga:users' } ],
+	};
+
+	const pieArgs = {
+		...baseArgs,
+		compareStartDate,
+		compareEndDate,
+		dimensions: [ dimensionName ],
+		orderby: {
+			fieldName: 'ga:users',
+			sortOrder: 'DESCENDING',
+		},
+		limit: 6,
+	};
+
+	const graphArgs = {
+		...baseArgs,
+		dimensions: [ 'ga:date' ],
+	};
+
+	const totalsArgs = {
+		...baseArgs,
+		compareStartDate,
+		compareEndDate,
+	};
+
+	if ( entityURL ) {
+		pieArgs.url = entityURL;
+		graphArgs.url = entityURL;
+		totalsArgs.url = entityURL;
+	}
+
+	if ( dimensionName && dimensionValue ) {
+		graphArgs.dimensionFilters = { [ dimensionName ]: dimensionValue };
+		totalsArgs.dimensionFilters = { [ dimensionName ]: dimensionValue };
+	}
+
+	const pieLoaded = useSelect( ( select ) => select( STORE_NAME ).hasFinishedResolution( 'getReport', [ pieArgs ] ) );
+	const pieError = useSelect( ( select ) => select( STORE_NAME ).getErrorForSelector( 'getReport', [ pieArgs ] ) );
+	const pieReport = useSelect( ( select ) => select( STORE_NAME ).getReport( pieArgs ) );
```

Comment:
Bit of a nitpick, but can we use slightly-more verbose variable names for these? Maybe `pieChartLoaded` and `userCountGraph`, etc.? Just to make things a bit clearer.

Reply:
Updated.